app/javascript/src/ChangePassword/components/ChangePasswordHooks.jsx

```diff
@@ -1,87 +1,95 @@
+// @flow
+
 import { useState } from 'react'
-import { validateForm, validateSingleField } from 'LoginPage/utils/formValidation'
+import validate from 'validate.js' // TODO: including this since validateSingleField and validateForm from LoginPage/utils/formValidation screams for a refactor
+
+type PasswordField = 'user[password]'
+type PasswordConfirmationField = 'user[password_confirmation]'
+type FieldName = PasswordField | PasswordConfirmationField
+type Constraints = {[FieldName]: {[string]: {}}}
```

Comment:
This could be more specific:
```
type Constraint = 'presence' | 'length' | 'equality'
type Constraints = { [FieldName]: { [Constraint]: {} } }

Reply:
Yep, it could definitely be more specific. Will change